Patch Author:  Orson Teodoro <orsonteodoro@hotmail.com>
Date: Jan 11, 2021

Limits tests if using only 128-bit AES only (aka i686 patch).

This allows comprehensive tests to continue and reports if if test passed.
Results viewable via dmesg.

----
diff -urp linux-4.14.214-ot.orig/crypto/testmgr.c linux-4.14.214-ot/crypto/testmgr.c
--- linux-4.14.214-ot.orig/crypto/testmgr.c	2021-01-12 12:08:29.095722672 -0800
+++ linux-4.14.214-ot/crypto/testmgr.c	2021-01-12 12:14:31.093740730 -0800
@@ -1185,8 +1185,17 @@ static int __test_skcipher(struct crypto
 						  CRYPTO_TFM_REQ_WEAK_KEY);
 
 #ifdef CONFIG_CRYPTO_TRESOR
-			if (strstr(algo, "tresor"))
+			if (strstr(algo, "tresor")) {
+				if (template[i].klen != 16) {
+					pr_info("alg: skcipher%s: klen=%d for %s is skipped and unsupported.",
+						d, template[i].klen, algo);
+					continue;
+				}
+
+				pr_info("alg: skcipher%s: testing %d on %s: klen=%d for %s.",
+					d, j, e, template[i].klen, algo);
 				tresor_setkey(template[i].key);
+			}
 #endif
 
 		ret = crypto_skcipher_setkey(tfm, template[i].key,
@@ -1245,6 +1254,13 @@ static int __test_skcipher(struct crypto
 			ret = -EINVAL;
 			goto out;
 		}
+
+#ifdef CONFIG_CRYPTO_TRESOR
+		if (strstr(algo, "tresor")) {
+			pr_info("alg: skcipher%s: passed test %d on %s: klen=%d for %s.",
+				d, j, e, template[i].klen, algo);
+		}
+#endif
 	}
 
 	j = 0;
@@ -1271,8 +1287,17 @@ static int __test_skcipher(struct crypto
 						  CRYPTO_TFM_REQ_WEAK_KEY);
 
 #ifdef CONFIG_CRYPTO_TRESOR
-			if (strstr(algo, "tresor"))
+			if (strstr(algo, "tresor")) {
+				if (template[i].klen != 16) {
+					pr_info("alg: skcipher%s: klen=%d for %s is skipped and unsupported.",
+						d, template[i].klen, algo);
+					continue;
+				}
+
+				pr_info("alg: skcipher%s: testing chunk test %d on %s: klen=%d for %s.",
+					d, j, e, template[i].klen, algo);
 				tresor_setkey(template[i].key);
+			}
 #endif
 
 		ret = crypto_skcipher_setkey(tfm, template[i].key,
@@ -1369,6 +1394,13 @@ static int __test_skcipher(struct crypto
 			}
 			temp += template[i].tap[k];
 		}
+
+#ifdef CONFIG_CRYPTO_TRESOR
+		if (strstr(algo, "tresor")) {
+			pr_info("alg: skcipher%s: passed chunk test %d on %s: klen=%d for %s.",
+				d, j, e, template[i].klen, algo);
+		}
+#endif
 	}
 
 	ret = 0;
